token/core/fabtoken/validator.go

```diff
@@ -212,23 +250,30 @@ func (v *Validator) VerifyTransfer(inputTokens []*token2.Token, tr driver.Transf
 			return errors.Wrapf(err, "failed parsing quantity [%s]", out.Quantity)
 		}
 		outputSum.Add(q)
+		// check that all outputs have the same type and it is the same type as inputs
 		if out.Type != typ {
 			return errors.Errorf("output type %s does not match type %s", out.Type, typ)
 		}
 	}
+	// check equality of sum of inputs and outputs
 	if inputSum.Cmp(outputSum) != 0 {
 		return errors.Errorf("input sum %v does not match output sum %v", inputSum, outputSum)
 	}
 	return nil
 }
 
 type backend struct {
-	getState   driver.GetStateFnc
-	message    []byte
-	index      int
+	getState driver.GetStateFnc
+	// signed message
+	message []byte
+	index   int
+	// signatures on message
 	signatures [][]byte
 }
 
+// HasBeenSignedBy checks if a given message has been signed by the signing identity matching
+// the passed verifier
+// todo shall we remove id from the parameters
```

Comment:
todo

Reply:
refactoring to be handled in different PR